libs/api/domains/auth/src/lib/services/types.ts

```diff
@@ -0,0 +1,88 @@
+import { User } from '@island.is/auth-nest-tools'
+import {
+  CreateDelegationInput,
+  DelegationInput,
+  DelegationsInput,
+  DeleteDelegationInput,
+  PatchDelegationInput,
+  UpdateDelegationInput,
+} from '../dto'
+import { DelegationByOtherUserInput } from '../dto/delegationByOtherUser.input'
+import { ApiScope } from '../models'
+import { ScopeTreeNode } from '../models/scopeTreeNode.model'
+import { ApiScopesInput } from '../dto/apiScopes.input'
+
+export interface MeDelegationsServiceI {
```

Comment:
Why ending with I? Version 1?

Reply:
I think it is redundant to use prefix or suffix on interfaces with ´I´. Thats more the C# way. 
We get proper type and context from things like intellesense and language servers.

But in the end it is just a preference :)

Reply:
Yeah. I agree, that's what I meant and what I strive for, but in this case I couldn't really find a way around it.

This is my problem:

```
// The interface
interface MeDelegationsService {}

// The main wrapper service
class MeDelegationsService implements MeDelegationsService {}

// Feature flagged service.
class MeDelegationsServiceV1 implements MeDelegationsService {}
class MeDelegationsServiceV2 implements MeDelegationsService {}
```

The interface and the main service can't have the same name in the same file, so I need to differentiate them somehow. Elsewhere, I want to use dependency injection to inject "MeDelegationsService", the actual class instead of the interface. And I kind of do not want to name it something like "MeDelegationsServiceWrapper". So it is better to rename the interface ... and "Interface" is the logical differentiator.

"The exception makes the rule"? Not happy about this, but here we are :D